Source/WebCore/page/Page.cpp

```diff
@@ -3954,7 +3954,7 @@ bool Page::useDarkAppearance() const
     RefPtr localMainFrame = dynamicDowncast<LocalFrame>(mainFrame());
     RefPtr view = localMainFrame ? localMainFrame->view() : nullptr;
     if (!view || view->mediaType() != screenAtom())
-        return false;
+        return m_useDarkAppearance;
```

Comment:
I don't think we should be changing how printing works here (`view->mediaType() != screenAtom()`). Presumably you only wanted to change the case where RemoteFrame was used?

Reply:
You are right. I changed the code to return `m_useDarkAppearance` if `localMainFrame` is nullptr.

I checked the rest of the function and I found it may return `m_useDarkAppearanceOverride` if it is set. And it can true if `DocumentLoader::colorSchemePreference() == ColorSchemePreference::Dark`.

1. `m_useDarkAppearanceOverride` is set from the `InspectorPageAgent` and `InspectorPageHost`.
2. colorSchemePreference is set from the `WKWebPagePreferences` and `WebsitePolicesData`.

So I think none of these should affect the appearance of the cross origin frames.

Reply:
I don't think printing is implemented yet.

It looks like printing iterates over all descendant frames to set the mediaType, but only does so for in-process/LocalFrames currently. That would need to work cross-process.

Once that is true, then we should be able to look at the set of 'root frames' (which are always LocalFrames), and check the mediaType on one of those. They should always be the same (at least for printing), unsure what happens if they aren't...

Reply:
Printing in site isolation is broken. The cross origin `<iframe>` is not exported to PDF. 
- I tried to get the `effectiveUseDarkAppearance()` of the main frame from the cross origin WebContent process but this breaks things because it requires two sync messages from WebProcess -> UIProcess -> WebProcess. And this affects other messages which are running at the same time like `layerTreeAsTextForTesting`.
- I tried to know whether we are printing from the cross origin WebContent process but it seems this information was not propagated.
- I added a layout test for printing but I marked it as` [ Failure ]` and I added a `FIXME` comment in `Page::effectiveUseDarkAppearance()`

Because the printing is broken fixing the dark mode for printing is outside the scope of fixing this bug.